diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-06-21 21:04:34.677563207 +0200
+++ b/CMakeLists.txt	2025-06-21 21:05:18.807623308 +0200
@@ -8,8 +8,7 @@
 add_library(${PROJECT_NAME} INTERFACE)
 target_include_directories(${PROJECT_NAME} INTERFACE "include")
 
-add_subdirectory("tests")
 if (false)
     add_subdirectory("example")
     add_subdirectory("benchmark")
-endif()
\ No newline at end of file
+endif()
